### PR TITLE
[ticket/12426] Allow search by custom profile fields

### PR DESCRIPTION
**Pull request by satanasov**
This is resubmit of PR #6287 

PHPBB3-12426

Checklist:

- [ ] Correct branch: master for new features; 3.3.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [ ] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-12426

### CODE REVIEW

**Comment by 3D-I** [comment]
Save your files with Unix/Mac CR/LF format

File: phpBB/memberlist.php [hunk 1]
```diff
@@ -1091,6 +1109,11 @@
 
 			$find_key_match = array('lt' => '<', 'gt' => '>', 'eq' => '=');
 
+			// Time to add aditional search parameters
+			foreach ($search_profilefields_params as $search_parameter)
+			{
+				$search_container[$search_parameter['field_ident']] = request_var($search_parameter['field_ident'], $search_parameter['field_novalue'], $search_parameter['field_multibyte']);
+			}
 			$find_count = array('lt' => $user->lang['LESS_THAN'], 'eq' => $user->lang['EQUAL_TO'], 'gt' => $user->lang['MORE_THAN']);
```
Comment (3D-I) [inline phpBB/memberlist.php:1117]: Short array syntax pls

File: phpBB/memberlist.php [hunk 2]
```diff
@@ -1433,6 +1485,10 @@
 			'ip'			=> array('ip', ''),
 			'first_char'	=> array('first_char', ''),
 		);
+		foreach ($search_profilefields_params as $additional_search_params)
+		{
+			$check_params[$additional_search_params['field_ident']] = array($additional_search_params['field_ident'], (isset($search_container[$additional_search_params['field_ident']])) ? $search_container[$additional_search_params['field_ident']] : '');
+		}
 
 		$u_first_char_params = array();
```
Comment (3D-I) [inline phpBB/memberlist.php:1493]: Short array syntax pls

File: phpBB/phpbb/profilefields/manager.php [hunk 3]
```diff
@@ -343,7 +343,7 @@ public function generate_profile_fields_template_headlines($restrict_option = ''
 		 * @var array	profile_cache	A copy of the profile cache to make additional checks
 		 * @since 3.1.6-RC1
 		 */
-		$vars = ['restrict_option', 'tpl_fields', 'profile_cache'];
+		$vars = array('restrict_option', 'tpl_fields', 'profile_cache');
```
Comment (3D-I) [inline phpBB/phpbb/profilefields/manager.php:346]: Short array syntax pls

File: phpBB/phpbb/profilefields/manager.php [hunk 4]
```diff
@@ -715,4 +715,111 @@ private function list_profilefields($type_name, $active = false)
...
+		return $custom_search_array;
+	}
+
+	/**
+	* Build and define templates for search form
+	*/
+	public function generate_search_fields()
+	{
+		// Lets get fields
+		$sql_array = array(
```
Comment (3D-I) [inline phpBB/phpbb/profilefields/manager.php:767]: Short array syntax pls - all over

File: phpBB/phpbb/profilefields/type/type_bool.php [hunk 5]
```diff
@@ -243,6 +243,89 @@ public function generate_field($profile_row, $preview_options = false)
...
+
+		if ($profile_row['field_length'] == 1)
+		{
+			if (!$this->lang_helper->is_set($profile_row['field_id'], $profile_row['lang_id'], 1))
+			{
+				$this->lang_helper->load_option_lang($profile_row['field_id']);
+			}
+
+			$options = $this->lang_helper->get($profile_row['field_id'], $profile_row['lang_id']);
+			$this->template->assign_block_vars('bool.options', array(
```
Comment (3D-I) [inline phpBB/phpbb/profilefields/type/type_bool.php:277]: Short array syntax pls - all over

File: phpBB/phpbb/profilefields/type/type_date.php [hunk 6]
```diff
@@ -273,6 +273,36 @@ public function generate_field($profile_row, $preview_options = false)
...
+		];
+		return $output;
+	}
+
+	/**
+	* {@inheritDoc}
+	*/
+	public function make_sql_where($profile_row, $db, $table_prefix = 'pd')
+	{
+		//@todo needs implementing
```
Comment (3D-I) [inline phpBB/phpbb/profilefields/type/type_date.php:302]: todos should be solved

File: phpBB/phpbb/profilefields/type/type_string.php [hunk 7]
```diff
@@ -115,6 +115,51 @@ public function generate_field($profile_row, $preview_options = false)
...
+
+		$this->template->assign_block_vars($this->get_name_short(), array_change_key_case($profile_row, CASE_UPPER));
+	}
+
+	/**
+	* {@inheritDoc}
+	*/
+	public function get_search_array($profile_row)
+	{
+		$output = array(
```
Comment (3D-I) [inline phpBB/phpbb/profilefields/type/type_string.php:136]: Short array syntax pls - all over

File: phpBB/phpbb/profilefields/type/type_text.php [hunk 8]
```diff
@@ -119,6 +119,55 @@ public function generate_field($profile_row, $preview_options = false)
...
+
+		$this->template->assign_block_vars('text', array_change_key_case($profile_row, CASE_UPPER));
+	}
+
+	/**
+	* {@inheritDoc}
+	*/
+	public function get_search_array($profile_row)
+	{
+		$output = array(
```
Comment (3D-I) [inline phpBB/phpbb/profilefields/type/type_text.php:144]: Short array syntax pls - all over

**Review by 3D-I — CHANGES_REQUESTED** [review]

File: phpBB/phpbb/profilefields/manager.php (hunk 3 above)
Comment (satanasov) [reply phpBB/phpbb/profilefields/manager.php:346]: Checks are erring with array missing for $var on event.

**Comment by rxu** [comment]
Mind to add some functional tests?

**Comment by satanasov** [comment]
When I have some more free time will do so.

**Pull request closed without merging**